workbench-core/datasets/src/wbcDataSetsAuthorizationPlugin.ts

```diff
@@ -17,7 +26,52 @@ export class WbcDataSetsAuthorizationPlugin implements DataSetsAuthorizationPlug
   }
 
   public async addAccessPermission(params: AddRemoveAccessPermissionRequest): Promise<PermissionsResponse> {
-    throw new Error('Method not implemented.');
+    if (params.permission.accessLevel !== 'read-only' && params.permission.accessLevel !== 'read-write') {
+      throw new InvalidPermissionError('Access Level must be "read-only" or "read-write".');
+    }
+
+    if (params.permission.identityType !== 'GROUP' && params.permission.identityType !== 'USER') {
+      throw new InvalidPermissionError('IdentityType just be "GROUP" or "USER".');
+    }
+
+    const permissionsEffect: Effect = 'ALLOW';
+    const permissionsAction: Action = params.permission.accessLevel === 'read-only' ? 'READ' : 'UPDATE';
+    const subjectType: IdentityType = params.permission.identityType === 'GROUP' ? 'GROUP' : 'USER';
```

Comment:
If `permission.identityType` was typed to `IdentityType` from the start we could avoid this mapping

Reply:
If typescript was more than just javascript under the hood, we could avoid this mapping.